src/test/java/com/groupon/jenkins/DockerComposeDotCi/DockerPublishPluginTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2016, Groupon, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+package com.groupon.jenkins.DockerComposeDotCi;
+
+import com.groupon.jenkins.buildtype.util.shell.ShellCommands;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.logging.Logger;
+
+public class DockerPublishPluginTest {
+
+    Logger LOGGER = Logger.getLogger(DockerPublishPlugin.class.getName());
+    ShellCommands cmds = new ShellCommands();
+    String composeBuildName = "htriantafilloumycoolapp42";  // GitHub org + repo + build number
+    String composeImage = "app";
+    String dotCiTag = "0.34.42";
+    String sha = "66630f67158d025cea4be1b2c8a80205e1136c87";  // echo meow | sha1sum
+    String dockerRepoName = "docker.domain.tld/htriantafillou/my_cool_app";
+    DockerPublishUtils utils = new DockerPublishUtils();
+
+    @Test
+    public void should_publish_sha_as_label_if_tag_is_null() {
+        ShellCommands final_cmds = utils.publishImages(cmds, composeBuildName,
+                composeImage, null, sha, dockerRepoName, false, LOGGER);
+
+        String cmd_lines[] = final_cmds.toShellScript().split("\\r?\\n");
+
+        String tag_cmd = cmd_lines[1];
+        String push_cmd = cmd_lines[3];
+
+        Assert.assertTrue(tag_cmd.endsWith(sha));
+        Assert.assertTrue(push_cmd.endsWith(sha));
```

Comment:
I suggest expanding the assertion to verify the entire line in the shell commands. It's possible to return a line that ends with the sha but wouldn't actually perform the publish.  It would also serve as documentation of what the full output should look like.

`Assert.assertEquals(tag_cmd, "docker tag app docker.domain.tld/htriantafillou/my_cool_app: 66630f67158d025cea4be1b2c8a80205e1136c87")` or something